### PR TITLE
Make sure roottest tests ROOT being build, not ROOT in env

### LINKED ISSUE #7081: `cmake -Droottest=On` tests wrong ROOT build!
### Describe the bug
Configuring a build with a different ROOT build already in the environment causes CTest to test the wrong build. I wonder how many devs are not testing their changes but that of some completely unrelated build :-(

### Expected behavior
`cmake -Droottest=On` should have `ctest` test *that* build.

### To Reproduce
```
$ which root
$ mkdir ../1
$ cd ../1
$ . /home/axel/build/root/master/deb/bin/thisroot.sh
$ cmake -Droottest=On ../src
$ grep ROOTSYS CMakeCache.txt 
ROOTSYS:INTERNAL=/home/axel/build/root/master/deb
```

### Setup
1. ROOT version: master
2. Operating system: Ubuntu 20.04
3. How you obtained ROOT: I built it myself.

### PR DESCRIPTION
**Pull request by Axel-Naumann**
Fixes #7081

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -586,7 +586,7 @@ if(testing)
 
     if(IS_DIRECTORY ${roottestdir})
       file(REMOVE_RECURSE ${CMAKE_CURRENT_BINARY_DIR}/roottest)
-      add_subdirectory(${roottestdir} roottest)
+      add_subdirectory(${roottestdir} roottest-build)
```
Comment (oshadura) [inline CMakeLists.txt:589]: Why we have here `roottest-build`?

Comment (Axel-Naumann) [reply CMakeLists.txt:589]: It's the *build* directory that's passed as the second argument. And I was always confused why I checked out `roottest` into my build dir ;-) I prefer to have the directory name clarify it's the roottest build dir. I'm okay to drop this if you find this wrong!

Comment (oshadura) [reply CMakeLists.txt:589]: nono, I was just curious...Let's use it!

**Comment by oshadura** [comment]
@Axel-Naumann `projectroot.roottest-build.cling.parsing.roottest_cling_parsing_semicolon` failure looks 'real'?

**Comment by Axel-Naumann** [comment]
> failure looks 'real'?

Totally - debugging...

**Review by oshadura — APPROVED** [review]

**Comment by Axel-Naumann** [comment]
`CTestCustom.cmake` suppresses `roottest.cling.parsing` not `roottest-build.cling.parsing`. This looks fragile (also wrt Jenkins) so I dropped the rename of the build dir. Too bad.

**Pull request merged**